### DISCUSSION TITLE
Change Parent

### DISCUSSION THREAD

**Discussion by kostyrin** [microsoftgraph/msgraph-sdk-dotnet#2494] [Q&A]
I would like to know how to change parent for DriveItem ? I have an example. but seems to be not working
```
var newParentDriveItemRequest = new DriveItem { ParentReference = new ItemReference { Id = parentId } };
await GraphClient.Drives[driveId].Items[driveItemId].Request().UpdateAsync(newParentDriveItemRequest);
```
got `Bad Request`.
Thanks for any advice

**Comment by andrueastman** [comment]
Thanks for raising this @kostyrin 

Any chance the example at this link is useful? https://learn.microsoft.com/en-us/graph/api/driveitem-move?view=graph-rest-1.0&tabs=csharp

Can you confirm that the two items also belong to the same `drive`? I believe this call won't work if the drives of the parent is different.

**Reply by kostyrin** [reply]
I can't use latest version of library.
But when trying to send request via HttpProvider
```
var tempDriveItemRequest = new DriveItem { Name = notebookName, ParentReference = new ItemReference { Id = parentId }, Package = new() { Type = "oneNote" } };
var json = JsonConvert.SerializeObject(tempDriveItemRequest, new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore });
var pathUrl = $"{GraphClient.Users[ownerId].Drive.Items[itemId].RequestUrl}";
using var request = new HttpRequestMessage(new HttpMethod("PATCH"), pathUrl)
{
    Content = new StringContent(json, Encoding.UTF8, "application/json")
};
await GraphClient.AuthenticationProvider.AuthenticateRequestAsync(request);
using var httpResponseMessage = await GraphClient.HttpProvider.SendAsync(request);
var response = await httpResponseMessage.Content.ReadAsStringAsync();
```
and http request will be
```
https://graph.microsoft.com/v1.0/users/{userName}/drive/items/itemId
{"Package":{"Type":"oneNote"},"Name":"Notebook","ParentReference":{"Id":"{newParentId}"},"ODataType":""}

```
got this: `Invalid request`
more details:
```
<strong>Message:</strong>
Code: invalidRequest
Message: Invalid request
Inner error:
	AdditionalData:
	date: 2024-05-16T15:10:02
	request-id: d4430e77-9947-4cfc-b0bd-0bc45b7edfb4
	client-request-id: d4430e77-9947-4cfc-b0bd-0bc45b7edfb4
ClientRequestId: d4430e77-9947-4cfc-b0bd-0bc45b7edfb4
```
Note that there is the file on url `https://graph.microsoft.com/v1.0/users/{userId}/drive/items/{itemId}`
and parent is existed on `https://graph.microsoft.com/v1.0/me/drives/{driveId}/items/{itemId}`

**Reply by kostyrin** [reply]
also this is item has been created as onenote package in Notebooks folder. I am just trying to change parent
